File: Tema5_ProvaHib/src/main/kotlin/classes/Comarca.java
package classes;

import javax.persistence.*;
import java.util.Collection;

@Entity
public class Comarca {
    private String nomC;
    private String provincia;
    private Collection<Poblacio> poblacions;

    @Id
    @Column(name = "nom_c")
    public String getNomC() {
        return nomC;
    }

    public void setNomC(String nomC) {
        this.nomC = nomC;
    }

    @Basic
    @Column(name = "provincia")
    public String getProvincia() {
        return provincia;
    }

    public void setProvincia(String provincia) {
        this.provincia = provincia;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Comarca comarca = (Comarca) o;

        if (nomC != null ? !nomC.equals(comarca.nomC) : comarca.nomC != null) return false;
        if (provincia != null ? !provincia.equals(comarca.provincia) : comarca.provincia != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = nomC != null ? nomC.hashCode() : 0;
        result = 31 * result + (provincia != null ? provincia.hashCode() : 0);
        return result;
    }

    @OneToMany(mappedBy = "comarca")
    public Collection<Poblacio> getPoblacions() {
        return poblacions;
    }

    public void setPoblacions(Collection<Poblacio> poblacions) {
        this.poblacions = poblacions;
    }
}

File: Tema5_PostgreSQL_geo_ad/src/main/kotlin/classes/Comarca.kt
package classes
import java.io.Serializable
import classes.Poblacio
import classes.Comarca
import javax.persistence.*

@Entity
class Comarca: Serializable {
    companion object {
        private const val serialVersionUID: Long = 1
    }

    @get:Column(name = "nom_c")
    @get:Id
    var nomC: String? = null

    @get:Column(name = "provincia")
    @get:Basic
    var provincia: String? = null

    @get:OneToMany(mappedBy = "comarca")
    var poblacions: Collection<Poblacio>? = null
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val comarca = o as Comarca
        if (if (nomC != null) nomC != comarca.nomC else comarca.nomC != null) return false
        return if (if (provincia != null) provincia != comarca.provincia else comarca.provincia != null) false else true
    }

    override fun hashCode(): Int {
        var result = if (nomC != null) nomC.hashCode() else 0
        result = 31 * result + if (provincia != null) provincia.hashCode() else 0
        return result
    }
}